src/antsibull_nox/ee.py

```diff
@@ -0,0 +1,161 @@
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or
+# https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: 2025, Ansible Project
+
+"""
+Build execution environments for testing.
+"""
+
+from __future__ import annotations
+
+import argparse
+import dataclasses
+import os
+import subprocess
+from pathlib import Path
+from typing import Optional
+
+import yaml
+
+from antsibull_nox.ee_config_generator import ExecutionEnvironmentGenerator
+
+
+@dataclasses.dataclass()
+class Args:
+    path: Path
+    namespace: Optional[str] = None
+    name: Optional[str] = None
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-p", "--path", help="path to collection", type=Path)
+    parser.add_argument("-s", "--namespace", help="collection namespace")
+    parser.add_argument("-n", "--name", help="collection name")
+    return parser.parse_args()
+
+
+def check_galaxy_version(args: Args):
```

Comment:
Check out how `antsibull_nox.sessions.build_import_check`'s `build_import_check()` does this. Also this function should receive a path as its argument, and not the full `Args` object.

Reply:
Thanks for the pointer, @felixfontein I've been looking at it but would like to clarify my understanding please.

`build_import_check()` creates temporary files instead of editing the actual `galaxy.yml` right? Would it be better to import this like `from .sessions.build_import_check import add_build_import_check` or rewrite my `check_galaxy_version()` to behave in a similar fashion?

Reply:
> `build_import_check()` creates temporary files instead of editing the actual `galaxy.yml` right?

Yes. It creates a copy of the collection (`copy_collection(Path.cwd(), collection_dir)`) and then modifies galaxy.yml in there if necessary.

> Would it be better to import this like `from .sessions.build_import_check import add_build_import_check` or rewrite my `check_galaxy_version()` to behave in a similar fashion?

I'd simply copy the required code over for now. We can refactor it later so there's a common functionality.

Reply:
I extracted the code in #83.

Reply:
Ah, awesome. Thanks @felixfontein ! I appreciate your patience while I work through this. Just to let you know I'm going to mostly be away until early next week but will make finishing this off a priority when I get back. It has been a good experience getting to integrate some code into your project so cheers again for that.